src/riscv_private.h

```diff
@@ -137,3 +148,33 @@ static inline uint32_t sign_extend_b(const uint32_t x)
 {
     return (int32_t) ((int8_t) x);
 }
+
+#if RV32_HAS(EXT_F)
+#include <math.h>
+#include "softfloat.h"
+
+#if defined(__APPLE__)
```

Comment:
This part is no longer needed.

Reply:
Except for Apple, the libraries of other computer architectures has provided the isinff and isnanf functions

Reply:
The following patch works for my Apple M1 machine:
```diff
--- a/src/riscv.c
+++ b/src/riscv.c
@@ -168,17 +168,6 @@ void rv_reset(riscv_t *rv, riscv_word_t pc)
     rv->halt = false;
 }
 
-#if defined(__APPLE__)
-static inline int isinff(float x)
-{
-    return __builtin_fabsf(x) == __builtin_inff();
-}
-static inline int isnanf(float x)
-{
-    return x != x;
-}
-#endif
-
 /* get current time in microsecnds and update csr_time register */
 static inline void update_time(riscv_t *rv)
 {
@@ -153,11 +153,15 @@ static inline uint32_t sign_extend_b(const uint32_t x)
 #include <math.h>
 #include "softfloat.h"
 
-#if defined(__APPLE__)
-inline int isinff(float x);
-inline int isnanf(float x);
+static inline int isinff(float x)
+{
+    return __builtin_fabsf(x) == __builtin_inff();
+}
+static inline int isnanf(float x)
+{
+    return x != x;
+}
 #endif
-#endif /* RV32_HAS(EXT_F) */
 
 #if RV32_HAS(Zicsr)
 /* CSRRW (Atomic Read/Write CSR) instruction atomically swaps values in the
```

Reply:
It only works for the Apple architecture. In an x86 machine, the error occurs.
```
src/riscv_private.h:156:19: error: static declaration of 'isinff' follows non-static declaration
static inline int isinff(float x)
                  ^
src/riscv_private.h:160:19: error: static declaration of 'isnanf' follows non-static declaration
static inline int isnanf(float x)
                  ^
```

Reply:
> It only works for the Apple architecture. In an x86 machine, the error occurs.

Consolidate the conditional build upon system configurations.